### PR TITLE
Simple VRAM Allocator

### PR DESCRIPTION
**Pull request by MrAwesome**
Replicate the logic of the previous allocator, but in safe code using a struct. 

Right now the alloc functions return a unique chunk object with the start point and the length. This would almost certainly look more complex in a more robust allocator, since the allocator would likely want to keep track of the allocated chunks for itself, or have the chunks store references to other chunks. Happy to make this look any way that makes sense, this is all uncharted territory to me but has been a lot of fun to hack on.

### CODE REVIEW

File: psp/src/lib.rs [hunk 1]
```diff
@@ -31,7 +31,7 @@ pub mod debug;
 #[macro_use] mod vfpu;
 mod eabi;
 pub mod sys;
-pub mod gu_utils;
```
Comment (overdrivenpotato) [inline psp/src/lib.rs:34]: Looks like the `gu_utils.rs` file wasn’t removed

File: psp/src/vram_alloc.rs [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+        self.offset += size;
+
+        if self.offset > self.total_mem() {
+            panic!("Total VRAM size exceeded!");
+        }
+
+        VramMemChunk::new(old_offset, size)
+    }
+
+    pub fn dealloc(&mut self, _chunk: VramMemChunk) {
```
Comment (overdrivenpotato) [inline psp/src/vram_alloc.rs:45]: No need for this function or `realloc` anymore

File: psp/src/vram_alloc.rs [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+    }
+
+    fn total_mem(&self) -> u32 {
+        unsafe {
+            sceGeEdramGetSize()
+        }
+    }
+}
+
+// TODO: can we hardcode SCREEN_WIDTH and SCREEN_HEIGHT here, or is there a use-case
```
Comment (MrAwesome) [inline psp/src/vram_alloc.rs:67]: Do you think we should leave this as width/height, or use SCREEN_WIDTH/SCREEN_HEIGHT? I'm not clear enough on the usecase for this one to know if it's likely to be used for smaller portions of the screen or not.

Comment (overdrivenpotato) [reply psp/src/vram_alloc.rs:67]: Let's leave it as width and height, this allows support for loading textures into VRAM with arbitrary sizes

**Comment by MrAwesome** [comment]
Alright! Should be good to go.

**Comment by overdrivenpotato** [comment]
Thanks!

**Pull request merged**